BUILD.md

```diff
@@ -0,0 +1,86 @@
+# Important notice
+***In order to uniformize and simplify the build system we had to make choices. TC is currently only officially supported on Ubuntu 16.04 with gcc 5.4.0, cuda 9.0 and cudnn 7.***
+Other configurations may work too but are not yet officially supported.
+For more information about setting up the config that we use to build the conda dependencies see the following [Dockerfile](conda_recipes/Dockerfile).
+
+Our main goal with this decision is to make the build procedure extremely simple, reproducible both internally and extensible to new targets in the future.
+In particular the gcc-4 / gcc-5 ABI switch is not something we want to concern ourselves with at this point, we go for gcc-5.4.0.
+
+# Conda from scratch (one time config)
+Choose and set a CONDA_PATH then run the following:
+
+```
+wget https://repo.anaconda.com/archive/Anaconda3-5.1.0-Linux-x86_64.sh -O anaconda.sh && \
+    chmod +x anaconda.sh && \
+    ./anaconda.sh -b -p ${CONDA_PATH} && \
+    rm anaconda.sh
+
+. ${CONDA_PATH}/bin/activate
+conda update -y -n base conda
+```
+
+Create a new environment in which TC will be built and install core dependencies:
+```
+conda create -y --name tc_build python=3.6
+conda activate tc_build
+conda install -y pyyaml mkl-include pytest
+conda install -y -c pytorch pytorch torchvision cuda90
+conda remove -y cudatoolkit --force
```

Comment:
Why do we remove `cudatoolkit`?  Can't we use it instead of the system one (I have 9.1 by default rather than 9.0, and will pass to 9.2 soon).

Reply:
You should have 9.0 for now, you can easily install a 9.0 toolkit and make /usr/local/cuda point to the right location. Once we get PyTorch + Caffe2 binaries that statically link cuda libraries we can go ahead and drop this.